frontend/public/components/masthead-toolbar.jsx

```diff
@@ -39,25 +39,18 @@ class MastheadToolbar_ extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.flags !== prevProps.flags) {
+    if (this.props.flags[FLAGS.OPENSHIFT] !== prevProps.flags[FLAGS.OPENSHIFT]
+      || !_.isEqual(this.props.user, prevProps.user)) {
       this._updateUser();
     }
   }
 
   _updateUser() {
-    const { flags } = this.props;
-
-    if (flags[FLAGS.OPENSHIFT]) {
-      coFetchJSON('api/kubernetes/apis/user.openshift.io/v1/users/~')
-        .then(user => {
-          this.setState({ username: _.get(user, 'fullName') || user.metadata.name });
-        })
-        .catch(() => {
-          this.setState({ username: null });
-        });
-    } else {
-      this.setState({ username: authSvc.userID() ? authSvc.name() : null });
+    const { flags, user } = this.props;
+    if (!flags[FLAGS.OPENSHIFT]) {
+      this.setState({ username: authSvc.name() });
```

Comment:
We probably want to skip this if not `AUTH_ENABLED`

Reply:
So no username should be shown at all if auth is disabled?

Reply:
Sorry, I'm wrong. Only the logout action is disabled if auth is disabled.